lib/services/azuresqldbfailovergroup/cmd-provision.js

```diff
@@ -163,26 +163,12 @@ var sqldbfgProvision = function (params) {
   };
 
   // validators
-  function isValidStringArray(a) {
-    // SQL injection detect
-    var regex = /^([ a-zA-Z0-9_-]+)$/;
-    if (!_.isArray(a)) {
-      return false;
-    } else {
-      var valid = true;
-      a.forEach(function(v){
-        if (!_.isString(v) || !regex.test(v)) valid = false;
-      });
-      return valid;
-    }
-  }
-
   this.areUserRolesValid = function () {
-    return isValidStringArray(userRoles);
+    return common.isValidStringArray(userRoles);
```

Comment:
Do we have a list of all allowed `userRoles` and `userPermissions`? Now we only make sure element in the array is a valid string, what if the user provides a wrong string?

Reply:
Binding would fail if the role or permission is not existed. The permission set is huge: https://docs.microsoft.com/en-us/sql/t-sql/statements/grant-database-permissions-transact-sql?view=sql-server-2017. And I don't know whether there is new patch for SQL Server which can add new permisssions or not. Is it a good idea to cache the list in broker side? With the updatable ability, I think it is acceptable if the user specify a invalid role or permission.

Reply:
It makes sense.